Online: Include project root path

theotherjimmy opened the pull request:
### Description

The prior logic assumed that "." would not be added to the include
paths, indicating that the project root would not be added to the 
include paths correctly in the online environment ("." would be 
incorrect there). This change set started by removing the addition 
of ".", and then fixed building from there.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

theotherjimmy committed:
Online: Include project root path

### Description

The prior logic assumed that "." would not be added to the include
paths, indicating that the project root would not be added to the
include paths correctly in the online environment ("." would be
incorrect there). This change set started by removing the addition
of "*.", and then fixed building from there.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

cmonr reviewed (approved):
Neat

cmonr commented:
Out of curiosity, why was the prior logic needed to begin with? Seems like this new behavior is more generic to begin with.

theotherjimmy commented:
@cmonr I think it was using `os.path.join`, which does not accept 0 arguments, instead of `self._sep.join`, which does.

NirSonnenschein commented:
/morph build

0xc0170 commented:
One failure (nothing uploaded to s3, can't locate the exact failure at the moment) - build aborted

theotherjimmy commented:
The true problem, in case you were wondering:
```
-I -I. -I./cmsis 
```

`-I` without an argument will consume `-I.`, which prevents includes from the root. I'll have to investigate why this is happening.

cmonr commented:
Hmm, would encapsulating the parameters in quotes help, or are you specifically trying to avoid that?

theotherjimmy committed:
Resources: Compute parents using only header names

### Description

The prior fix made the assumption that you wanted to compute all of the
parents for a give header file going all the way up the path. This is
not true: you probably want to stop when the project stops. We already
keep track of a virtual name within the project, so instead, we compute
parents of the name, and generate the actual location of these files in
your FS as the path. This makes the solution robust offline and online
(I tested it with my local copy of os.mbed.com)

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

theotherjimmy commented:
@cmonr Good guess, but that would not help much/at all. The problem was computing parents past the project's root directory. I sorted that with the new commit.

cmonr commented:
**Note**: This PR is now a part of a rollup PR (#8686).

No further work is needed here, as once that PR is merged, this PR will also be closed and marked as merged.

_If any more commits are made in this PR, this PR will remain open and have to go through CI on its own._
